static/js/user_status.js

```diff
@@ -42,6 +45,44 @@ export function revoke_away(user_id) {
     away_user_ids.delete(user_id);
 }
 
+// This function will add missing/extra parameters to the emoji info object,
+// that would need by template to render an emoji.
+export function get_emoji_info(emoji_info) {
+    // To call this function you must pass at least an emoji name.
+    if (!emoji_info || !emoji_info.emoji_name) {
+        return {};
+    }
+
+    const status_emoji_info = {...emoji_info};
+
+    status_emoji_info.emoji_alt_code = page_params.emojiset === "text";
+    if (status_emoji_info.emoji_alt_code) {
+        return status_emoji_info;
+    }
+
+    if (emoji.active_realm_emojis.has(emoji_info.emoji_name)) {
+        if (!emoji_info.reaction_type) {
+            if (emoji_info.emoji_name === "zulip") {
+                status_emoji_info.reaction_type = "zulip_extra_emoji";
+            } else {
+                status_emoji_info.reaction_type = "realm_emoji";
+            }
+        }
+        const more_emoji_info = emoji.active_realm_emojis.get(emoji_info.emoji_name);
+        status_emoji_info.emoji_code = emoji_info.emoji_code || more_emoji_info.id;
+        status_emoji_info.url = more_emoji_info.emoji_url;
+    } else {
+        const codepoint = emoji.get_emoji_codepoint(emoji_info.emoji_name);
+        if (codepoint === undefined) {
+            blueslip.warn("Bad emoji name: " + emoji_info.emoji_name);
+            return {};
+        }
+        status_emoji_info.reaction_type = emoji_info.reaction_type || "unicode_emoji";
+        status_emoji_info.emoji_code = emoji_info.emoji_code || codepoint;
+    }
+    return status_emoji_info;
+}
```

Comment:
This function basically duplicates a function from `reactions.js`; I think we should try to move that parsing logic to a new `emoji.get_emoji_by_name` that can be called from both code paths and returns an object with all these properties.  

Can you do that rework?  It doesn't look like this is every called with any other parameters, so that seems cleaner.

Reply:
Added the new commit refactoring this, though instead of naming the function `get_emoji_by_name` I named it `get_emoji_details_by_name` as there already exists a function with the name `get_emoji_name`,

https://github.com/zulip/zulip/blob/779a2f33171a345d18971dccc61fc31ae395ba3a/static/shared/js/emoji.js#L76